assert/assertions.go

```diff
@@ -76,6 +76,48 @@ func ObjectsAreEqual(expected, actual interface{}) bool {
 	return bytes.Equal(exp, act)
 }
 
+// ObjectsExportedFieldsAreEqual determines if the exported (public) fields of two structs are considered equal.
+// If the two objects are not of the same type, or if either of them are not a struct, they are not considered equal.
+//
+// This function does no assertion of any kind.
```

Comment:
🤔 Given that this is the assert package, would it not make sense if we did assert?

Right now the functionality of this is more of a helper function, should we not just change it to an assert function that asserts whether `object_a`'s exported fields == `object_b`'s exported fields?

Reply:
We have another function that is added further down in the file called `EqualExportedValues` that actually does the assertion by calling this `ObjectsExportedFieldsAreEqual` helper function.

We thought that it would make sense to split up the logic for actually doing the comparison (and put it into a helper function) from the actual assertion function. This seems to be the approach that is used for equality comparisons with a `ObjectsAreEqual` helper function that is then called by the `Equal` function that actually does the assertion.